test/index.ts

```diff
@@ -249,4 +271,99 @@ describe('teeny', () => {
       });
     });
   });
+
+  it('should track stats, callback mode, success', done => {
+    const scope = mockJson();
+    teenyRequest({uri}, () => {
+      assert.ok(statsInstanceStubs.requestStarting.calledOnceWithExactly());
+      assert.ok(statsInstanceStubs.requestFinished.calledOnceWithExactly());
+      scope.done();
+      done();
+    });
+  });
+
+  it('should track stats, callback mode, failure', done => {
+    const scope = mockError();
+    teenyRequest({uri}, err => {
+      assert.ok(err);
+      assert.ok(statsInstanceStubs.requestStarting.calledOnceWithExactly());
+      assert.ok(statsInstanceStubs.requestFinished.calledOnceWithExactly());
+      scope.done();
+      done();
+    });
+  });
+
+  it('should track stats, stream mode, success', done => {
+    const scope = mockJson();
+    const readable = teenyRequest({uri});
+    assert.ok(statsInstanceStubs.requestStarting.calledOnceWithExactly());
+
+    readable.once('response', () => {
+      assert.ok(statsInstanceStubs.requestFinished.calledOnceWithExactly());
+      scope.done();
+      done();
+    });
+  });
+
+  it('should track stats, stream mode, failure', done => {
+    const scope = mockError();
+    const readable = teenyRequest({uri});
+    assert.ok(statsInstanceStubs.requestStarting.calledOnceWithExactly());
+
+    readable.once('error', err => {
+      assert.ok(err);
+      assert.ok(statsInstanceStubs.requestFinished.calledOnceWithExactly());
+      scope.done();
+      done();
+    });
+  });
+
+  // TODO multipart is broken with 2 strings
```

Comment:
I would add a link to the feature request ticket you open here, for adding multipart support.

Reply:
opened issue https://github.com/googleapis/teeny-request/issues/168
also added the link next to this TODO